functions-and-operators/window-functions.md

```diff
@@ -8,22 +8,466 @@ aliases: ['/docs/dev/functions-and-operators/window-functions/','/docs/dev/refer
-| [`ROW_NUMBER()`](https://dev.mysql.com/doc/refman/8.0/en/window-function-descriptions.html#function_row-number)| Returns the number of the current row in the partition. |
+| Function name                     | Feature description |
+| :-------------------------------- | :------------------------------------- |
+| [`CUME_DIST()`](#cume_dist)       | Returns the cumulative distribution of a value within a group of values. |
+| [`DENSE_RANK()`](#dense_rank)     | Returns the rank of the current row within the partition, and the rank is without gaps. |
+| [`FIRST_VALUE()`](#first_value)   | Returns the expression value of the first row in the current window. |
+| [`LAG()`](#lag)                   | Returns the expression value from the row that precedes the current row by N rows within the partition. |
+| [`LAST_VALUE()`](#last_value)     | Returns the expression value of the last row in the current window. |
+| [`LEAD()`](#lead)                 | Returns the expression value from the row that follows the current row by N rows within the partition. |
+| [`NTH_VALUE()`](#nth_value)       | Returns the expression value from the N-th row of the current window. |
+| [`NTILE()`](#ntile)               | Divides a partition into N buckets, assigns the bucket number to each row in the partition, and returns the bucket number of the current row within the partition. |
+| [`PERCENT_RANK()`](#percent_rank) | Returns the percentage of partition values that are less than the value in the current row. |
+| [`RANK()`](#rank)                 | Returns the rank of the current row within the partition. The rank may be with gaps. |
+| [`ROW_NUMBER()`](#row_number)     | Returns the number of the current row in the partition. |
+
+## [`CUME_DIST()`](https://dev.mysql.com/doc/refman/8.0/en/window-function-descriptions.html#function_cume-dist)
+
+`CUME_DIST()` calculates the cumulative distance. Note that the window definition should use `ORDER BY`. Otherwise this will not return the expected values.
+
+```sql
+WITH RECURSIVE cte(n) AS (
+    SELECT 1
+    UNION
+    SELECT
+        n+2
+    FROM
+        cte
+    WHERE
+        n<6
+)
+SELECT
+    *,
+    CUME_DIST() OVER(ORDER BY n)
+FROM
+    cte;
+```
+
+```
++------+------------------------------+
+| n    | CUME_DIST() OVER(ORDER BY n) |
++------+------------------------------+
+|    1 |                         0.25 |
+|    3 |                          0.5 |
+|    5 |                         0.75 |
+|    7 |                            1 |
++------+------------------------------+
+4 rows in set (0.00 sec)
+```
+
+## [`DENSE_RANK()`](https://dev.mysql.com/doc/refman/8.0/en/window-function-descriptions.html#function_dense-rank)
+
+The `DENSE_RANK()` function is similar to [`RANK()`](#rank) but does not leave any gaps in case of ties.
```

Comment:
```suggestion
The `DENSE_RANK()` function returns the rank of the current row. It is similar to [`RANK()`](#rank) but does not leave any gaps in case of ties (rows that share the same values and order conditions).
```